File: src/main/java/my/study/repository/design/pattern/factory/java/EasyFactory.java
package my.study.repository.design.pattern.factory.java;

/**
 * @author djh on  2018/7/28 23:08
 * @E-Mail 1544579459@qq.com
 */
public class EasyFactory {
    public static BMW getBMWThree() {
        return new BMWThree();
    }

    public static BMW getBMWSix() {
        return new BMWSix();
    }
}

File: src/main/java/my/study/repository/design/pattern/factory/kotlin/EasyFactory.kt
package my.study.repository.design.pattern.factory.kotlin

/**
 * @author djh on  2018/10/5 18:56
 * @E-Mail 1544579459@qq.com
 */
class EasyFactory {
    companion object {
        
        fun getBmwSix() = BMWSix(68)
        
        fun getBmwThree() = BMWThree(33)
    }
}



